components/pipeline-service/production/base/main-pipeline-service-configuration.yaml

```diff
@@ -1504,7 +1504,7 @@ spec:
         - -check_owner=false
         - -completed_run_grace_period=5m
         - -requeue_interval=2m
-        - -store_deadline=90m
+        - -store_deadline=240m
```

Comment:
For a paper trail on why this is necessary: since there is an order of magnitude more TaskRuns than PipelineRuns, TaskRuns may take 60-120 minutes to be reconciled after they're complete, whereas their corresponding PipelineRun may be reconciled in 20-30 minutes after completion. When this happens, the PipelineRun reconciler sees that the PipelineRun has been complete for 30, 60, or 90 minutes but its TaskRuns have not yet been stored. It then deletes the pipelinerun without storing because its passed the "store_deadline". This change allows the PipelineRun reconciler to wait longer for its TaskRuns to be stored/deleted before giving up on the PipelineRun.

Reply:
The store_deadline applies to finished PipelineRuns, which means at that time all TaskRuns should be finished as well. From that point we wait 4 hours for everything to be stored before we delete the objects without a "stored" annotation. Obviously if objects are stored earlier, they will be also pruned earlier.